lib/query.js

```diff
@@ -2950,7 +2956,7 @@ function completeOne(model, doc, res, options, fields, userProvidedFields, pop,
     return null;
   }
 
-  const casted = helpers.createModel(model, doc, fields, userProvidedFields);
+  const casted = helpers.createModel(model, doc, fields, userProvidedFields, options);
```

Comment:
This generally looks right, but maybe should be `this._mongooseOptions` instead of `options`?

Reply:
for some reason it throws an error stating cannot read property mongooseOptions of undefined